scripts/Dockerfile.e2e

```diff
@@ -0,0 +1,76 @@
+# syntax=docker/dockerfile:1.7-labs
+# (use non-stable syntax for convenient --parents option in COPY command)
+
+# Download clang
+FROM scratch AS clang
+ADD https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.4/clang+llvm-16.0.4-x86_64-linux-gnu-ubuntu-22.04.tar.xz /clang.tar.xz
+
+FROM ubuntu:latest
+
+#####################
+# DEPENDENCIES
+#####################
+# Install misc build dependencies
+# TODO: only install required hard requirements, see best practices
+# https://docs.docker.com/build/building/best-practices/#apt-get
+RUN apt-get update && \
+    apt-get install -y -qq \
```

Comment:
Is there a reason to not do a --no-install-recommends here?

Reply:
I have a branch with `--no-install-recommends` and with quite a few deps removed: https://github.com/lukpueh/lind-wasm/blob/273d8ba44f274ae3f5c633f3b7c0849c60ca6d5f/Dockerfile#L13-L32

But I didn't have time yet to confirm that it doesn't break anything.